sdk/go/pulumi/context.go

```diff
@@ -1032,14 +1231,14 @@ func getTimeouts(custom *CustomTimeouts) *pulumirpc.RegisterResourceRequest_Cust
 
 // getOpts returns a set of resource options from an array of them. This includes the parent URN, any dependency URNs,
 // a boolean indicating whether the resource is to be protected, and the URN and ID of the resource's provider, if any.
-func (ctx *Context) getOpts(t string, providers map[string]ProviderResource, opts *resourceOptions, remote bool,
-) (URN, []URN, bool, string, map[string]string, bool, ID, []string, []string, string, error) {
+func (ctx *Context) getOpts(t string, provider ProviderResource, opts *resourceOptions, remote bool,
+) (URN, []URN, bool, string, map[string]string, bool, ID, []string, []string, error) {
```

Comment:
10-tuples remind me of Haskell pain from previous job, sitting there counting out positions to the 7th argument during a 3am debugging session. Could we use a record to name the fields? I can undertake a separate PR 😅